contractcourt/chain_arbitrator.go

```diff
@@ -137,10 +137,9 @@ type ChainArbitratorConfig struct {
 	// Sweeper allows resolvers to sweep their final outputs.
 	Sweeper *sweep.UtxoSweeper
 
-	// SettleInvoice attempts to settle an existing invoice on-chain with
-	// the given payment hash. ErrInvoiceNotFound is returned if an invoice
-	// is not found.
-	SettleInvoice func(lntypes.Hash, lnwire.MilliSatoshi) error
+	// Registry is the invoice database that is used by resolvers to lookup
+	// preimages and settle invoices.
+	Registry *invoices.InvoiceRegistry
```

Comment:
Rather than giving them the entire struct, how about introducing a new interface that specifies _only_ what the resolvers need? This will make writing unit tests easier as well, since they don't need to fully instantiate an `InvoiceRegistry`.

Reply:
I could do that, but don't see too much benefit at this moment.